bitflip/bitflip.rb

```diff
@@ -0,0 +1,105 @@
+#!/usr/bin/env ruby
+
+require 'net/http'
+require 'resolv'
+
+def pad_with_zeros(string)
+	padded = string
+	until padded.length == 8
+		padded = "0" + padded
+	end
+	return padded
+end
+
+def is_valid_domain_char(original, char)
+	return false if char == original - 32
+	return (65 <= char && char <= 90) || (97 <= char && char <= 122) || (char == 45)
+end
+
+def is_valid_domain(domain, tlds)
+	return false if (domain.size > 255 || domain.size < 4)
+	return false if !domain.include?(".")
+	domain.split(".").each do |chunk|
+		return false if (chunk[0] == "-" || chunk[chunk.size] == "-")
+	end
+	if tlds != nil
+		return false if !tlds.include?(domain.split(".")[-1])
+	end
+	return true
+end
+
+def download_tlds
+	tlds = nil
+	puts "Downloading current list of top level domains..."
```

Comment:
haha we can use my [list](https://github.com/jborrey/securityTools/blob/master/dns/tlds.txt) now.